### PR TITLE
fix duplicated kafka consumer ID in global hub agent.

### PR DESCRIPTION
**Pull request by morvencao**
fix: https://github.com/stolostron/backlog/issues/27083
which is introduced in https://github.com/stolostron/multicluster-global-hub/pull/259/

Signed-off-by: morvencao <lcao@redhat.com>

### CODE REVIEW

File: operator/pkg/controllers/addon/manifests/templates/agent/multicluster-global-hub-agent-deployment.yaml [hunk 1]
```diff
@@ -24,6 +24,7 @@ spec:
             - '--zap-devel=true'
             - --pod-namespace=$(POD_NAMESPACE)
             - --leaf-hub-name={{ .LeafHubID }}
+            - --kakfa-consumer-id={{ .LeafHubID }}
```
Comment (clyang82) [inline operator/pkg/controllers/addon/manifests/templates/agent/multicluster-global-hub-agent-deployment.yaml:27]: what is default value without specifying it obviously?

Comment (yanmxa) [reply operator/pkg/controllers/addon/manifests/templates/agent/multicluster-global-hub-agent-deployment.yaml:27]: Each consumer has a unique consumer.id across its consumer group. If you don't define a consumer.id, the Kafka client (Java, Python, Node.js, etc) usually chooses a random ID.
Refer: https://stackoverflow.com/questions/34550873/difference-between-groupid-and-consumerid-in-kafka-consumer

**Comment by yanmxa** [comment]
I think maybe we need to make sure different regional hub have a different `group.id`. 
image

As you can see from this figure, if two consumers are in the same group, there is no guarantee that each consumer in the group will receive the message, but every group will receive the full information about the topic.

File: operator/pkg/controllers/addon/manifests/templates/agent/multicluster-global-hub-agent-deployment.yaml (hunk 1 above)
Comment (morvencao) [reply operator/pkg/controllers/addon/manifests/templates/agent/multicluster-global-hub-agent-deployment.yaml:27]: we use `kakfa-consumer-id` as kafka consumer ID and group ID, see: https://github.com/stolostron/multicluster-global-hub/blob/7dcb96485c9576e26d8a875134cb1a095e87b17e/pkg/transport/consumer/kafka_consumer.go#L69-L70 That means each agent can receive all the partitions of the message.

**Comment by morvencao** [comment]
we use `kakfa-consumer-id` as kafka consumer ID and group ID, see: https://github.com/stolostron/multicluster-global-hub/blob/7dcb96485c9576e26d8a875134cb1a095e87b17e/pkg/transport/consumer/kafka_consumer.go#L69-L70 That means each agent can receive all the partitions of the message.
@yanmxa

**Comment by clyang82** [comment]
It was introduced by https://github.com/stolostron/multicluster-global-hub/pull/259. the previous value is regional hub name. #259 made it as an immutable value.

/lgtm

**Pull request merged**